pallets/vector/src/lib.rs

```diff
@@ -841,8 +864,18 @@ pub mod pallet {
 			proof: ProofInput,
 			public_values: PublicValuesInput,
 		) -> DispatchResultWithPostInfo {
+			use substrate_bn_succinct::Fr;
 			let sender: [u8; 32] = ensure_signed(origin)?.into();
-			let updater = Updater::<T>::get();
+			let pico_updater = PicoUpdater::<T>::get();
+			let sp1_updater = Updater::<T>::get();
+			let (updater, updater_type) = if H256(sender) == pico_updater {
```

Comment:
Not an issue but style-wise would a match operator be more elegant in this instance?